bucket/diagram-designer.json

```diff
@@ -0,0 +1,19 @@
+{
+    "version": "1.29.5",
+    "description": "Vector graphics editor for creating flowcharts, UML class diagrams, illustrations and slide shows",
+    "homepage": "https://logicnet.dk/DiagramDesigner/",
+    "license": "Freeware",
+    "notes": "Template palettes can be downloaded from 'https://github.com/meesoft/DiagramDesigner/tree/master/TemplatePalettes'",
+    "url": "https://www.fosshub.com/Diagram-Designer.html/DiagramDesignerSetup.1.29.5.msi",
+    "hash": "381c696c81ab57faaf40ce690c760f549056fb1af299ebf550c08dae5f379bf7",
+    "shortcuts": [
+        [
+            "DiagramDesigner.exe",
+            "Diagram Designer"
+        ]
+    ],
+    "checkver": "Diagram Designer ([\\d.]+)",
+    "autoupdate": {
+        "url": "https://www.fosshub.com/Diagram-Designer.html/DiagramDesignerSetup.$version.msi"
+    }
```

Comment:
On https://www.fosshub.com/Diagram-Designer.html/, when you press the Signature button you can get access to the SHA256 hash of the file.

Would it be possible for us to extract that hash using existing tooling in Scoop?

That is if Diagram Designer is expected to get updates. By the looks of their GitHub repo it doesn't seem like that is going to ever be a case

Reply:
Currently it is implemented in *Scoop*'s main program (*FossHub mode*). Therefore we don't need to download the file while running *autoupdate* for files hosted on FossHub.
https://github.com/ScoopInstaller/Scoop/blob/092005046454d94d141d5c68fdbdb4c4a1229ae9/lib/autoupdate.ps1#L222.

I would say it is better to extend the feature and enable users to make use of it **in the manifest**, but this requires further update on *Scoop*'s main program.
I also wish that child paths (e.g. `some_app\x86\bin`) can be applied on *extract_dir* through all means of extraction including *innounp*, *lessmsi*, *uniextract2*, etc. , but that is another topic.

Reply:
I had no idea!!!